src/GovUk.Education.ExploreEducationStatistics.Notifier/Functions/SubscriptionManager.cs

```diff
@@ -254,7 +254,8 @@ public class SubscriptionManager
                 }
             }
 
-            return new BadRequestObjectResult("Unable to verify subscription");
+            return new RedirectResult(
+                $"{_appSettingOptions.PublicAppUrl}/subscriptions/verification-error");
```

Comment:
I'm not sure it's correct to use a redirect here - what's actually occurred? If the token has expired shouldn't we get a 403 or 401? Or if it's a bad payload maybe a 422? I'd prefer to return a 4xx status code and handle the problem in the frontend server.

Using a redirect will also have SEO implications - firstly we'd need to make sure it's a permeant not a temporary redirect, and secondly we'd then need to stop Google indexing the error verification page as the canonical. Perfectly doable, but feels like there are better options than using a redirect maybe?

Reply:
It was a 400. The issue was that the Notifier is a separate endpoint unrelated to the Public site on which we wanted to display the error page, so it can't be handled by middleware. `RedirectResult` is used in other partS of this class for subscription update/unsubscribe.